solutions/usaco-104.mdx

```diff
@@ -21,26 +21,71 @@ int main () {
 	int n, k;
 	cin >> n >> k;
 	vector<int> diff(n + 1);
-	for (int i = 0; i < k; i++) {
+	for (int i = 0; i < k; i++) { // read input and build difference array
 		int l, r;
 		cin >> l >> r;
-		l--;
+		l--; // make zero-indexed
 		r--;
 		diff[l]++;
 		diff[r + 1]--;
 	}
 
 	int sol[n];
 	int tot = 0;
-	for (int i = 0; i < n; i++) {
+	for (int i = 0; i < n; i++) { // build prefix sum array
 		tot += diff[i];
 		sol[i] = tot;
 	}
-	sort (sol, sol + n);
-	cout << sol[n/2] << '\n';
+	sort (sol, sol + n); // sort to get median
+	cout << sol[n/2] << '\n'; // output the median
 }
 
 ```
 
 </CPPSection>
+	
 </LanguageSection>
+<JavaSection>
+
+## Java Implementation
+
+```java
+import java.util.*;
+import java.io.*;
+
+class haybaleStacking {
+    public static void main (String[] args) {
+        Kattio io = new Kattio();
+
+        int n = io.nextInt();
+        int k = io.nextInt();
+        int[] differenceArray = new int[n + 1]; // differenceArray[n] is the difference between A[n + 1] and A[n]
+
+        for (int x = 0; x < k; x++) { // read input and build differenceArray
+            int a = io.nextInt();
+            int b = io.nextInt();
+            a--; // make zero-indexed
+            b--;
```

Comment:
why not combine into instantiation

Reply:
i'm not sure what that is, and google searches don't tell me much either

Reply:
just do 
```
int a = io.nextInt() - 1;
int b = io.nextInt() - 1;
```

Reply:
ok i did that and did i do the other changes right?

sry first time doing this